### PR TITLE
[tele5] Add new extractor

### LINKED ISSUE #7805: please support tele5.de
Can you please add support for the "Re-Play" pages of the german TV channel Tele 5?

Their video area starts at http://www.tele5.de/re-play.html - click on one of the "Jetzt anschauen" (or similar) buttons to go to an actual video (the individual URLs are only valid for about a week).

> # youtube-dl http://www.tele5.de/a-z/star-trek-das-naechste-jahrhundert/ganze-folgen/video/staffel-4-der-pakt-mit-dem-teufel.html
> [generic] staffel-4-der-pakt-mit-dem-teufel: Requesting header
> WARNING: Falling back on generic information extractor.
> [generic] staffel-4-der-pakt-mit-dem-teufel: Downloading webpage
> [generic] staffel-4-der-pakt-mit-dem-teufel: Extracting information
> ERROR: Unsupported URL: http://www.tele5.de/a-z/star-trek-das-naechste-jahrhundert/ganze-folgen/video/staffel-4-der-pakt-mit-dem-teufel.html

### PR DESCRIPTION
**Pull request by xflr6**
closes #7805

based on #7922, partially supersedes it (no `re-play` urls here)

example url not geo-blocked?

### CODE REVIEW

**Comment by dstftw** [comment]
This looks identical to kaltura extractor thus extraction should be delegated to it. Kaltura extractor may be generalized for custom api URLs if necessary.

**Comment by xflr6** [comment]
Can you point to the exact code lines you want to reuse here?

**Comment by dstftw** [comment]
Most likely none of them, extraction process is already well implemented and tested in kaltura extractor.

**Comment by xflr6** [comment]
I meant what exact lines from the other extractor you want to reuse in this extractor..

**Comment by dstftw** [comment]
You didn't get it. You should not reuse. You should [delegate](https://github.com/rg3/youtube-dl/blob/master/youtube_dl/extractor/ellentv.py#L81-L83). And extend kaltura extractor to work with external api URLs if necessary.

**Comment by xflr6** [comment]
Okay, return `url_result` with an added optional field for overriding `api_base` [here](https://github.com/rg3/youtube-dl/compare/master...xflr6:tele5-kaltura).

The backend does actually provide the API used by `KalturaIE._get_video_info` (though that is not used on the website AFAICS). However, the `video_url`s returned do not work:

```
$ python test/test_download.py TestDownload.test_Tele5
[Tele5] man-vs-fly/ganze-folgen/video/boxer: Downloading webpage
[Kaltura] 0_3vnqiwvl: Downloading Kaltura signature
[Kaltura] 0_3vnqiwvl: Downloading video info JSON
[Kaltura] 0_3vnqiwvl: Downloading m3u8 information
[Kaltura] 0_3vnqiwvl: Checking mp4-15486 video format URL
[Kaltura] 0_3vnqiwvl: Checking mp4-444 video format URL
[Kaltura] 0_3vnqiwvl: Checking mp4-631 video format URL
[Kaltura] 0_3vnqiwvl: Checking mp4-864 video format URL
[Kaltura] 0_3vnqiwvl: Checking 3gp-362 video format URL
[Kaltura] 0_3vnqiwvl: Checking hls-meta video format URL
[info] Writing video description metadata as JSON to: 0_3vnqiwvl.info.json
[debug] Invoking downloader on u'http://api.medianac.com/p/102/sp/10200/flvclipp
er/entry_id/0_3vnqiwvl/version/0/flavorId/0_jhtzjrgb'


ERROR: Did not get any data blocks
```

So similar but maybe not identical?

**Comment by Crypt0Graf** [comment]
I'm not sure, but I think you're doing it a little bit complicated. [The XML](http://api.medianac.com/p/102/sp/10200/playManifest/entryId/0_3vnqiwvl/format/rtmp/protocol/rtmp/uiConfId/11601158/a/a.f4m?referrer=aHR0cDovL3d3dy50ZWxlNS5kZQ) is  providing the base-URL ( _rtmp://rtmp.mnac-p-000000102.c.nmdn.net/mnac-p-000000102_ ) and the URLs of the media files in different qualities. If you're replacing _rtmp://rtmp.mnac-p-000000102.c.nmdn.net/mnac-p-000000102_ with _http://dl.mnac-p-000000102.c.nmdn.net/mnac-p-000000102_ and add the media file _mp4:20160307/0/0_3vnqiwvl_0_7xtta08k_2_ as _/20160307/0/0_3vnqiwvl_0_7xtta08k_2.mp4_ , you're ending up with _http://dl.mnac-p-000000102.c.nmdn.net/mnac-p-000000102/20160307/0/0_3vnqiwvl_0_7xtta08k_2.mp4_ . This is a direct mp4-link for the video. NOTE: This works most of the time, but sometimes (I had this some time ago) this doesn't work, so if you're implementing this method, please also add a fallback.

**Comment by remitamine** [comment]
may be using one of the api to extract the urls will make the extractor work for both kaltura and other based sites.
http://www.kaltura.com/api_v3/testmeDoc/index.php?service=flavorAsset&action=geturl or http://www.kaltura.com/api_v3/testmeDoc/index.php?service=flavorAsset&action=getdownloadurl.

**Comment by devurandom** [comment]
Will this be able to download e.g. http://www.tele5.de/re-play/spielfilme?v=5240 ?

**Comment by Crypt0Graf** [comment]
@devurandom Nope, probably not. Tele5 has completely rebuild their site, so it woudn't work after all. However, they are now shipping their content with an .mpd-info-file which links to .mp4-files. 
So in your case, http://tele5.akamaized.net/5240_c25ac40f/manifest.mpd links in the highest resolution to http://tele5.akamaized.net/5240_c25ac40f/A1003567_struppi_2400k_dashinit.mp4 .
The .mpd says, as far as I understand, that there are different files, which are addressed with relative paths ( means http://tele5.akamaized.net/5240_c25ac40f/ + filename ). Line 42 of the .mdp contains the syntax/filename of the highest resolution of the video `(    <BaseURL>A1003567_struppi_2400k_dashinit.mp4</BaseURL>)`. That's basically how we're getting the file path http://tele5.akamaized.net/5240_c25ac40f/A1003567_struppi_2400k_dashinit.mp4 .

Unfortunately, it looks like the audiocodec isn't recognised by every program; in my case, VLC wasn't able to play the audio track, Firefox however was able to play that file.

tl;dr: It should be easy to implement in yt-dl, but the audio codec could cause problems.
